pkg/operator/controller/garden/access/reconciler.go

```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package access
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/afero"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	kubernetesclient "github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operator/apis/config"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	"github.com/gardener/gardener/pkg/operator/controller/gardenlet"
+)
+
+// Reconciler reconciles garden access secrets.
+type Reconciler struct {
+	Client  client.Client
+	Manager manager.Manager
+	Config  *config.OperatorConfiguration
+
+	FS            afero.Fs
+	tokenFilePath string
+
+	GardenletControllerAdded bool
+}
+
+// Reconcile processes the given access secret in the request.
+// It extracts the included Kubeconfig, and prepares a dedicated REST config
+// where the inline Bearer Token is replaced by a BearerTokenFile with the said token persisted.
+// Afterward, the config is used to add the 'controller/gardenlet' controller to the manager.
+// Any subsequent reconciliation run causes the content of the BearerTokenFile to be updated with the token found
+// in the access secret.
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	secret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, request.NamespacedName, secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.V(1).Info("Object is gone, stop reconciling")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
+	}
+
+	kubeConfigBytes, ok := secret.Data[kubernetesclient.KubeConfig]
```

Comment:
```suggestion
	kubeConfigRaw, ok := secret.Data[kubernetesclient.KubeConfig]
```

Nit/often called like this